File: Diagram/CustomDraggingService/CustomDraggingService/CustomDraggingServiceForm.cs
using Telerik.Windows.Diagrams.Core;

namespace CustomDraggingService
{
    #region InitialSetup
    public partial class CustomDraggingServiceForm : Telerik.WinControls.UI.RadForm
    {
        private CustomDraggingService dragService;

        public CustomDraggingServiceForm()
        {
            InitializeComponent();

            this.dragService = new CustomDraggingService(this.radDiagram1.DiagramElement)
            {
                DragMode = DragMode.Horizontal
            };

            this.radDiagram1.DiagramElement.ServiceLocator.Register<IDraggingService>(this.dragService);
        }

        private void radButton1_Click(object sender, System.EventArgs e)
        {
            this.dragService.DragMode = this.dragService.DragMode == DragMode.Horizontal ? DragMode.Vertical : DragMode.Horizontal;
        }
    }
    #endregion
}

File: Diagram/CustomDraggingService/CustomDraggingServiceVB/CustomDraggingServiceForm.vb
Imports Telerik.Windows.Diagrams.Core

#Region "InitialSetup"
Public Class CustomDraggingServiceForm

    Dim dragService As CustomDraggingService

    Sub New()

        InitializeComponent()

        Me.dragService = New CustomDraggingService(Me.RadDiagram1.DiagramElement) With {.DragMode = DragMode.Horizontal}
        Me.RadDiagram1.DiagramElement.ServiceLocator.Register(Of IDraggingService)(Me.dragService)

    End Sub
    Private Sub RadButton1_Click(sender As Object, e As EventArgs) Handles RadButton1.Click
        Me.dragService.DragMode = If(Me.dragService.DragMode = DragMode.Horizontal, DragMode.Vertical, DragMode.Horizontal)
    End Sub
End Class

#End Region